eclass/cmake-utils.eclass

```diff
@@ -652,25 +652,67 @@ enable_cmake-utils_src_compile() {
 }
 
 _ninjaopts_from_makeopts() {
-	if [[ ${NINJAOPTS+set} == set ]]; then
+	if [[ ${NINJAOPTS+set} == set ]] ; then
```

Comment:
Because then this function is a function with side effects. Suppose someone would like to use this function no matter what is the current state of `NINJAOPTS` or do something else then exporting the result to NINJAOPTS.

I'd like to refactor this function to have API like this:
```
Usage: _ninjaopts_from_makeopts [ MAKEOPTS ]
Return: NINJAOPTS
```

Then this function wouldn't make any assumptions, have no side effects and would be generally reusable.

Also `_cmake_ninja_src_make()` would require only minimal changes, which would also made the code more readable.

now:
```ebuild
_ninjaopts_from_makeopts
```
then:
```ebuild
if [[ ${NINJAOPTS+set} != set ]] ; then
	export NINJAOPTS="$(_ninjaopts_from_makeopts)"
ii
```